Source/Fluids/WarpXFluidContainer.H

```diff
@@ -0,0 +1,172 @@
+        amrex::Real cur_time, bool skip_deposition=false);
+
+    /**
+     * AdvectivePush_Muscl takes a single timestep (dt) of the cold relativistic fluid equations
+     * using a Muscl-Handcock scheme
+     */
+    /**
+     * \brief Advective term, cold-rel. fluids
+     *
+     * \param[in] lev refinement level
+     */
+    void AdvectivePush_Muscl (int lev);
+
+
+    /**
+    * Apply (non-periodic) BC on the fluids (needed for spatial derivative),
+    * and communicate N, NU at boundaries
+    */
+    /**
+     * \brief Apply non-periodic BC to fluids and communicate boundaries
+     *
+     * \param[in] lev refinement level
+     */
+    void ApplyBcFluidsAndComms (int lev);
+
+    /**
+     * centrifugal_source adds contributions due to curvature acceleration for a
+     * single timestep using an SSP-RK3 timestep
+    */
+    /**
+     * \brief Centrifugal source term
+     *
+     * \param[in] lev refinement level
+     */
+    void centrifugal_source (int lev);
+
+    /**
+     * GatherAndPush introduces the lorentz term in the cold relativistic fluid
+     * equations for a single timestep (dt) using Higuera and Cary Push
+     */
+    /**
+     * \brief Lorentz Momentum Source
+     *
+     * \param[in] lev refinement level
+     * \param[in] Ex Yee electric field (x)
+     * \param[in] Ey Yee electric field (y)
+     * \param[in] Ez Yee electric field (z)
+     * \param[in] Bx Yee magnetic field (x)
+     * \param[in] By Yee magnetic field (y)
+     * \param[in] Bz Yee magnetic field (z)
+     * \param[in] cur_time Current time
+     */
+    void GatherAndPush (int lev,
+        const amrex::MultiFab& Ex, const amrex::MultiFab& Ey, const amrex::MultiFab& Ez,
+        const amrex::MultiFab& Bx, const amrex::MultiFab& By, const amrex::MultiFab& Bz,
+        amrex::Real cur_time);
+
+    /**
+     * DepositCurrent interpolates the fluid current density comps. onto the Yee grid and
+     * sums the contributions to the particle current density
+     */
+    /**
+     * \brief Deposit fluid current density.
+     *
+     * \param[in] lev refinement level
+     * \param[in,out] jx current density MultiFab x comp.
+     * \param[in,out] jy current density MultiFab y comp.
+     * \param[in,out] jz current density MultiFab z comp.
+     */
+    void DepositCurrent (int lev,
+        amrex::MultiFab& jx, amrex::MultiFab& jy, amrex::MultiFab& jz);
+
+    /**
+     * DepositCharge deposits density onto rho
+     */
+    /**
+     * \brief Deposit fluid charge density.
+     *
+     * \param[in] lev refinement level
+     * \param[in,out] rho charge density MultiFab.
+     */
+    void DepositCharge (int lev, amrex::MultiFab &rho, int icomp = 0);
+
+    amrex::Real getCharge () const {return charge;}
+    amrex::Real getMass () const {return mass;}
+
+protected:
+    int species_id;
+    std::string species_name;
+    amrex::Real charge;
+    amrex::Real mass;
+
+    int do_not_push = 0;
```

Comment:
Currently, this variable is read/initialized in `WarpXFluidContainer.cpp`, but actually never used. We should implement the logic that corresponds to this variable (or remove it).

Reply:
Logic implemented.